filewatcher/filewatcher.go

```diff
@@ -308,6 +305,26 @@ func New(ctx context.Context, cfg Config) (*Result, error) {
 	if err != nil {
 		return nil, err
 	}
+	isDir, err := isDirectory(cfg.Path)
+	if err != nil {
+		return nil, err
+	}
+	if isDir {
+		f = DummyReadCloser{
+			Path: cfg.Path,
```

Comment:
if Read or Close gets called on this, it will panic... so this doesn't feel right to me.  What is this trying to do?  `f` doesn't seem to be used after this

Reply:
If I understand correctly, this dummy read closer was added, so that we can pass through the path using an `io.Reader` interface due to the limitation of the existing API.

`f` is being used down here https://github.com/reddit/baseplate.go/pull/472/files#diff-eb266ac0bad4d099e00f9c659d61679d726029ef44c28bb064647bf4e9d1dc7dR330

If I read this correctly, the watcher loop will then call it and pass in the dummy again where it is being type asserted: https://github.com/reddit/baseplate.go/pull/472/files#diff-c6bb5cecf85ab156298ad011c2a0fd07ecb8bc37ec2f3dc06923703b44f6cbd2R83

Reply:
Yes thats what its for. Because you cant make io.Readers from directories I needed to have this extra object to support it. The entire filewatcher is built around io.readers and Im not sure how to support directories without a rework of the filewatcher or a new class altogether.

Reply:
What is not OK is leaving this around to panic.  You should _implement_ Read and Close.  They should return an error that says "this operation is not supported on a directory" rather than panic'ing.

(but see the comment below about an alternate API proposal)

Reply:
+1 on that :slightly_smiling_face:

Reply:
Im on it